--- plugins/access.fs/class.fsAccessDriver.php.orig	2012-05-09 10:54:16.000000000 +0000
+++ plugins/access.fs/class.fsAccessDriver.php	2012-05-09 11:17:17.000000000 +0000
@@ -500,13 +500,6 @@
 				AJXP_Logger::debug("Upload Files Data", $fileVars);
 				$destination=$this->urlBase.AJXP_Utils::decodeSecureMagic($dir);
 				AJXP_Logger::debug("Upload inside", array("destination"=>$destination));
-				if(!$this->isWriteable($destination))
-				{
-					$errorCode = 412;
-					$errorMessage = "$mess[38] ".SystemTextEncoding::toUTF8($dir)." $mess[99].";
-					AJXP_Logger::debug("Upload error 412", array("destination"=>$destination));
-					return array("ERROR" => array("CODE" => $errorCode, "MESSAGE" => $errorMessage));
-				}	
 				foreach ($fileVars as $boxName => $boxData)
 				{
 					if(substr($boxName, 0, 9) != "userfile_") continue;
@@ -1286,12 +1279,8 @@
 		{
 			return "$mess[40]"; 
 		}
-		if(!$this->isWriteable($this->urlBase."$crtDir"))
-		{
-			return $mess[38]." $crtDir ".$mess[99];
-		}
 
-        $dirMode = 0775;
+		$dirMode = 0775;
 		$chmodValue = $this->repository->getOption("CHMOD_VALUE");
 		if(isSet($chmodValue) && $chmodValue != "")
 		{
@@ -1303,6 +1292,10 @@
 		$old = umask(0);
 		mkdir($this->urlBase."$crtDir/$newDirName", $dirMode);
 		umask($old);
+		if(!file_exists($this->urlBase."$crtDir/$newDirName"))
+		{
+			return $mess[38]." $crtDir ".$mess[99];
+		}
 		return null;		
 	}
 	
@@ -1317,10 +1310,7 @@
 		{
 			return "$mess[71]";
 		}
-		if(!$this->isWriteable($this->urlBase."$crtDir"))
-		{
-			return "$mess[38] $crtDir $mess[99]";
-		}
+
 		$fp=fopen($this->urlBase."$crtDir/$newFileName","w");
 		if($fp)
 		{
@@ -1333,12 +1323,16 @@
 			}
 			$this->changeMode($this->urlBase."$crtDir/$newFileName");
 			fclose($fp);
-			return null;
 		}
 		else
 		{
 			return "$mess[102] $crtDir/$newFileName (".$fp.")";
-		}		
+		}
+                if(!file_exists($this->urlBase."$crtDir/$newFileName"))
+                {
+                        return "$mess[38] $crtDir $mess[99]";
+                }
+		return null;
 	}
 	
 	
