diff --git a/edit_ssl.cgi b/edit_ssl.cgi
index a8b6274..f552dbb 100755
--- a/edit_ssl.cgi
+++ b/edit_ssl.cgi
@@ -261,19 +261,15 @@ print ui_tabs_end_tab();
 
 # Let's Encrypt form
 print ui_tabs_start_tab("mode", "lets");
-my $err = &check_letsencrypt();
+my $err = 1;
 print $text{'ssl_letsdesc'};
 if (!$err) {
-	print &ui_tag('span',
-		&ui_details({
-			'class' => 'inline inlined',
-			'title' => '',
-			'content' => $text{'ssl_letsdesc2'},
-			}))."\n".
-		&ui_tag('style',
-			".ui--span>details.inline>summary+span {\n".
-			"margin-top: 0;\n".
-			"}\n");
+    print "<b>Unfortunately the Webmin Let's Encrypt module currrently clashes";
+    print " with TurnKey's SSL conf and has been disabled</b><p>\n";
+    print "<p>To use Let's Encrypt, please use <a href=";
+    print "'https://www.turnkeylinux.org/docs/confconsole'>Confconsole</a>.";
+    print " - see <a href='https://www.turnkeylinux.org/docs/confconsole/letsencrypt'>";
+    print "Let's Encrypt plugin</a> docs.</p>";
 	}
 print "<p>\n";
 
