diff --git a/edit_ssl.cgi b/edit_ssl.cgi
index dd98182..1a4cd77 100755
--- a/edit_ssl.cgi
+++ b/edit_ssl.cgi
@@ -259,13 +259,14 @@ print ui_tabs_end_tab();
 print ui_tabs_start_tab("mode", "lets");
 print "$text{'ssl_letsdesc'}<p>\n";
 
-my $err = &check_letsencrypt();
+my $err = 1;
 if ($err) {
-	print "<b>",&text('ssl_letserr', $err),"</b><p>\n";
-	print &get_letsencrypt_install_message(
-		"/$module_name/edit_ssl.cgi?mode=lets", $text{'ssl_title'});
-	print "<p>\n";
-	print &text('ssl_letserr2', "../config.cgi?$module_name"),"<p>\n";
+    print "<b>Unfortunately the Webmin Let's Encrypt module currrently clashes";
+    print " with TurnKey's SSL conf and has been disabled</b><p>\n";
+    print "<p>To use Let's Encrypt, please use <a href=";
+    print "'https://www.turnkeylinux.org/docs/confconsole'>Confconsole</a>.";
+    print " - see <a href='https://www.turnkeylinux.org/docs/confconsole/letsencrypt'>";
+    print "Let's Encrypt plugin</a> docs.</p>";
 	}
 else {
 	# Show form to create a cert
